docs/content/guides/dagster/branch_depl_with_snowflake.mdx

```diff
@@ -0,0 +1,325 @@
+---
+title: Branch Deployments with Snowflake | Dagster
+description: This guide illustrates a workflow that enables testing Dagster code in your cloud environment without impacting your production data.
+---
+
+# Branch Deployments with Snowflake
+
+This guide details a workflow to test Dagster code in your cloud environment without impacting your production data. To highlight this functionality, we’ll leverage Dagster Cloud branch deployments and a Snowflake database to:
+
+- Execute code on a feature branch directly on Dagster Cloud
+- Read and write to a clone of our Snowflake data
+
+With these tools, we can merge changes with confidence in the impact on our data platform and with assurance that our code will execute as intended.
+
+### Prerequisites
+
+<Note>
+  This guide is an extension of the{" "}
+  <a href="https://docs.dagster.io/guides/dagster/local-to-production-tutorial">
+    Transitioning Data Pipelines from Development to Production
+  </a>{" "}
+  guide, illustrating a workflow for staging deployments. We’ll use the examples
+  from this guide to build a workflow atop Dagster Cloud’s branch deployment
+  feature.
+</Note>
+
+This guide utilizes GitHub actions to trigger deployment—you’ll need to be using GitHub as your version control platform. You’ll also need to follow the [branch deployment setup guide](https://docs.dagster.cloud/guides/branch-deployments) and clone the [branch deployments quickstart template](https://github.com/dagster-io/dagster-cloud-branch-deployments-quickstart). After doing this, you should have:
+
+- A running agent to serve your branch deployments
+- A Dagster project set up for branch deployments containing the following:
+  - `branch_deployments.yml` GitHub workflow file
+  - Dagster repository (we’ve named ours `repo`)
+  - Dockerfile that installs your Dagster project
+
+If you’re new to Dagster, here’s an overview of the main concepts we’ll be using:
+
+- [Asset](https://docs.dagster.io/concepts/assets/software-defined-assets) - an asset is a software object that models a data asset. The prototypical example is a table in a database or a file in cloud storage.
```

Comment:
I think this section could be more powerful if, instead of giving explanations of these concepts that readers can already find on the concept pages, we explained how we intend to use them in this example.  E.g. for "job", we could say "We'll use a job that runs once per branch to set up a Snowflake database in our branch deployment".